File: Lista1Full.c
#include <stdio.h>
#include <stdlib.h>
#include <locale.h>

int p;
main()
{
    setlocale(LC_ALL, "");
    do{
    printf("Téstè\n");
    system("cls");
    printf("Este programa contem todos os exercícios da Lista 1 de Algoritmo\n");
    printf("Exercícios:\n");
    printf("EX 1 Antecessor/Sucessor                        EX 7 Calculo da Média Escolar\n");
    printf("EX 2 Reajuste +10%%                              EX 8 N > 50.000?             \n");
    printf("EX 3 Desconto de 9%%                             EX 9 São Multiplos?          \n");
    printf("EX 4 Salário Professor                          EX 10 Inflação de Produtos   \n");
    printf("EX 5 Carro Novo                                 EX 11 Maior Número           \n");
    printf("EX 6 Reajuste Salárial                          EX 12 Média + Conceito Escolar\n");
    printf("\nDigite o número do exercício: ");
    scanf("%d",&p);
    getchar();
    //
    if(p==1)
        ex1();
    else
    {
        if(p==2)
            ex2();
        else
        {
            if(p==3)
                ex3();
            else
            {
                if(p==4)
                    ex4();
                else
                {
                    if(p==5)
                        ex5();
                    else
                    {
                        if(p==6)
                            ex6();
                        else
                        {
                            if(p==7)
                                ex7();
                            else
                            {
                                if(p==8)
                                    ex8();
                                else
                                {
                                    if(p==9)
                                        ex9();
                                    else
                                    {
                                        if(p==10)
                                            ex10();
                                        else
                                        {
                                            if(p==11)
                                                ex11();
                                            else
                                            {
                                                if(p==12)
                                                    ex12();
                                                else
                                                    break;
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    }while(p==0);
}
ex1()
{
    setlocale(LC_ALL, "Portuguese");
    int a,s,x;
    char z;
    do{
    system("cls");
    printf("Este programa mostra o sucessor e o antecessor do número digitado.\n");
    printf("Digite o primeiro número: ");
    scanf("%d",&x);
    a=x-1;
    s=x+1;
    printf("O antecessor é: %d\n",a);
    printf("O sucessor é: %d\n",s);
    printf("\n\nDeseja fazer outra consulta? ('N' para sair)\n\t");
    scanf("%c", &z);
    z=getchar();
    }while ((z!='n') && (z!='N'));
    printf("\n\n\t\tFIM\n\n");
    p=0;
}
ex2()
{
    setlocale(LC_ALL, "Portuguese");
    float x;
    char z;
    do{
    system("cls");
    printf("Este programa reajusta um valor em 10%%\n");
    printf("Digite um valor para ser reajustado: ");
    scanf("%f",&x);
    printf("O valor reajustado é: %.2f\n",x*1.1);
    printf("\n\nDeseja fazer outra consulta? ('N' para sair)\n\t");
    scanf("%c", &z);
    z=getchar();
    }while ((z!='n') && (z!='N'));
    printf("\n\n\tFIM\n\n");
    p=0;
}
ex3()
{
    setlocale(LC_ALL, "Portuguese");
    float x;
    char z;
    do{
    system("cls");
    printf("Este programa desconta 9%% do preço de um produto.\n");
    printf("Digite o preço do produto: R$");
    scanf("%f",&x);
    while(x<=0)
    {
        printf("Valor Inválido!");
        printf("Digite o preço do produto: R$");
        scanf("%f",&x);
    }
    printf("O valor com desconto é: R$%.2f\n",x*.91);

    printf("\nDeseja fazer outra consulta? ('N' para sair)\n\t");
    scanf("%c", &z);
    z=getchar();
    }while ((z!='n') && (z!='N'));
    printf("\n\n\tFIM\n");
    p=0;
}
ex4()
{
        setlocale(LC_ALL, "Portuguese");
    float x,i;
    int n;
    char z;
    do{
    system("cls");
    printf("Este programa calcula o salário de professor.\n");
    printf("Insira o valor da Hora/Aula: R$");
    scanf("%f",&x);
    while(x<=0)
    {
        printf("Valor Inválido!");
        printf("Insira um valor maior que 0: R$");
        scanf("%f",&x);
    }
    printf("Insira a quantidade de aulas dadas: ");
    scanf("%d",&n);
    while(n<=0)
    {
        printf("Valor Inválido!");
        printf("Insira um valor maior que 0: ");
        scanf("%d",&n);
    }
    printf("Insira o percentual do INSS: ");
    scanf("%f",&i);
    while(i<=0)
    {
        printf("Valor Inválido!");
        printf("Insira um valor maior que 0: ");
        scanf("%f",&i);
    }
    printf("\nO salário é: R$%.2f",(x*n*(1-(i/100))));
    printf("\n\nDeseja fazer outra consulta? ('N' para sair)\n\t");
    scanf("%c", &z);
    z=getchar();
    }while ((z!='n') && (z!='N'));
    printf("\n\n\tFIM\n");
    p=0;
}
ex5()
{
    float vfab, imp, luc;
char zn;

    setlocale(LC_ALL, "");
    do{
    system("cls");
    printf("Este programa calcula o custo de um carro novo.\n");

    //Entrada Preço
    printf("Informe o preço de fábrica: R$");
    scanf("%f",&vfab);
    getchar();
    while(vfab<=0)
    {
        printf("Valor Inválido!\nInforme um valor maior que 0: R$");
        scanf("%f",&vfab);
        getchar();
    }

    //Entrada lucro
    printf("Informe o percentual de lucro: ");
    scanf("%f",&luc);
    getchar();
    while(luc<=0)
    {
        printf("Valor Inválido!\nInforme um valor maior que 0: ");
        scanf("%f",&luc);
        getchar();
    }

    //Entrada Lucro
    printf("Informe o percentual de impostos: ");
    scanf("%f",&imp);
    getchar();
    while(imp<=0)
    {
        printf("Valor Inválido!\nInforme um valor maior que 0: ");
        scanf("%f",&imp);
        getchar();
    }

    //Saída
    printf("O lucro do distribuidor é: R$%.2f",(vfab*(luc/100)));
    printf("\nO valor dos impostos é   : R$%.2f",(vfab*(imp/100)));
    printf("\nO valor final do carro é : R$%.2f",(vfab*(1+imp/100)*(1+luc/100)));

    printf("\n\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();

    }while ((zn != 'n') && (zn != 'N'));
   printf("\nfim \n");
    p=0;
}
ex6()
{
    float valor;
char zn;

    setlocale(LC_ALL, "Portuguese");
    do{
    system("cls");
    printf("Este programa calcula o reajuste do salário dependendo de seu valor.\n");
    printf("Insira o salário para reajuste: R$");
    scanf("%f", &valor);
    getchar();
    while(valor<=0)
    {
        printf("Valor Inválido!");
        printf("\nInsira um salário válido(>0): ");
        scanf("%f", &valor);
        getchar();
    }
    if(valor<800)
    {
        printf("De R$%.2f, o salário foi reajustado para R$%.2f.",valor,valor*1.15);
    }
    else
    {
        if(valor>1000)
        {
            printf("De R$%.2f, o salário foi reajustado para R$%.2f.",valor,valor*1.05);
        }
        else
        {
            printf("De R$%.2f, o salário foi reajustado para R$%.2f.",valor,valor*1.1);
        }
    }

    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();

    }while ((zn != 'n') && (zn != 'N'));
   printf("\nfim \n");
    p=0;
}
ex7()
{
    float n1, n2, n3, n4, ma;
char zn;

    setlocale(LC_ALL, "Portuguese");
    do{
    system("cls");
    printf("Este programa calcula a média das notas de 4 avaliações.\n");
   //Entrada de dados
   //N1
   printf("Insira a nota da Primeira Avaliação: ");
   scanf("%f", &n1);
   getchar();
    while((10 < n1) || (n1 < 0))
      {
        printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10: ");
         scanf("%f", &n1);
         getchar();
      }
   //N2
   printf("Insira a nota da Segunda Avaliação: ");
   scanf("%f", &n2);
   while((10<n2) || (n2<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10: ");
         scanf("%f", &n2);
         getchar();
      }
   //N3
   printf("Insira a nota da Terceira Avaliação: ");
   scanf("%f", &n3);
   while((10<n3) || (n3<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10: ");
         scanf("%f", &n3);
         getchar();
      }
   //n4
   printf("Insira a nota da Quarta Avaliação: ");
   scanf("%f", &n4);
   while((10<n4) || (n4<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10: ");
         scanf("%f", &n4);
         getchar();
      }
   //Processamento
   ma = ((n1+n2+n3+n4)/4);

   //Saida de dados
   printf("Média: %.2f\n", ma);
   if(ma>=6)
   {
      printf("Aprovado! \n");
   }
   else
   {
      printf("Reprovado! \n");
   }
    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    zn=getchar();

   }while ((zn != 'n') && (zn != 'N'));
   printf("\nfim \n");
    p=0;
}
ex8()
{
       float n1, n2, n3, n4, n5;
    char zn, x;

    setlocale(LC_ALL, "Portuguese");
    do{
            do{
    system("cls");
   //Entrada de dados
   //N1
   printf("Insira o Primeiro valor: ");
   scanf("%f", &n1);
   while ((25<=n1) || (n1<=10))
      {
        printf("Valor Invalido! \n");
         printf("Insira um valor maior que 10 e menor que 25: ");
         scanf("%f", &n1);
      }
      //N2
   printf("Insira o Segundo valor: ");
   scanf("%f", &n2);
   getchar();
    while(n2 < 0)
      {
        printf("Valor Invalido! \n");
         printf("Insira um valor maior que 0: ");
         scanf("%f", &n2);
         getchar();
      }
      //Processamento
      n3=n1+n2;
      n4=n1*n2*n3;
      n5=((n1*n1)+(n2*n2)+(n3*n3)+(n4*n4));
      if(n5<50000)
      {
          printf("\n\t(%.2f^2)+(%.2f^2)+(%.2f^2)+(%.2f^2) = %.2f < 50.000\n", n1,n2,n3,n4,n5);
          printf("Por favor insira novos valores.\n'ENTER' para continuar.");
          scanf("%c", &x);
      }
      else
      {
          printf("\n\t(%.2f^2)+(%.2f^2)+(%.2f^2)+(%.2f^2) = %.2f >= 50.000\n", n1,n2,n3,n4,n5);
          printf("Parabens! pressione 'ENTER' para sair.");
          scanf("%c", &x);

      }
    }while(n5<50000);
    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();

    }while ((zn != 'n') && (zn != 'N'));
    printf("\n\tFIM");
    p=0;
}
ex9()
{
    int a, b;
    char zn;
    setlocale(LC_ALL, "Portuguese");
    do{
    system("cls");
    printf("Este programa determina se dois valores são multiplos.\n");
    printf("Insira o primeiro número: ");
    scanf("%d", &a);
    getchar();
    printf("Insira o segundo número: ");
    scanf("%d", &b);
    getchar();
    if((a % b == 0)||(b % a == 0))
        {
        printf("\nOs números são multiplos!");
    }
    else
    {
        printf("\nOs números não são multiplos!");
    }
    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();
    }while ((zn != 'n') && (zn != 'N'));
    printf("\n\tFIM");
    p=0;
}
ex10()
{
    float valor;
    char zn;
    setlocale(LC_ALL, "Portuguese");
    //Loop
    do{
    system("cls");
    printf("Este programa calcula o preço reajustado de um produto.\n");
    //Entrada de dados
    printf("Insira o preço do produto: R$");
    scanf("%f", &valor);
    getchar();
    while(valor<=0)
    {
        printf("Preço Inválido");
        printf("Insira um valor maior que 0: R$");
        scanf("%f", &valor);
        getchar();
    }
    //Processamento
    if(valor<100)
        printf("\nPreço reajustado: R$%.2f",valor*1.1);
    else
        printf("\nPreço reajustado: R$%.2f",valor*1.2);
    //Loop ou fim
    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();
    }while ((zn != 'n') && (zn != 'N'));
    printf("\n\tFIM");
    p=0;
}
ex11()
{
//seção de Declarações das variáveis
int n1, n2;
char zn;

    setlocale(LC_ALL, "Portuguese");
    do{
    system("cls");
    printf("Este programa determina o maior de dois números\n");
    printf("Insira o primeiro número: ");
    scanf("%d",&n1);
    getchar();
    printf("Insira o segundo número : ");
    scanf("%d",&n2);
    getchar();
    if(n1>n2)
        printf("%d > %d",n1,n2);
    else
        if(n2>n1)
        printf("%d > %d",n2,n1);
    else
        printf("%d = %d",n1,n2);

    printf("\nDeseja repetir? (S/N)\n");
    scanf("%c", &zn);
    getchar();
    }while ((zn != 'n') && (zn != 'N'));
    printf("\n\tFIM");
    p=0;
}
ex12()
{
//seção de Declarações das variáveis
int id;
float n1, n2, n3, me, ma;
char x, zn;

    setlocale(LC_ALL, "Portuguese");
    do{
            system("cls");
   //Entrada de dados  ID
   printf("Digite sua Identificação:");
   scanf("%d", &id);
   getchar();

   //N1
   printf("Insira a nota da Primeira Avaliação:");
   scanf("%f", &n1);
   getchar();

    while((10 < n1) || (n1 < 0))
      {
        printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10:");
         scanf("%f", &n1);
         getchar();
      }


   //N2
   printf("Insira a nota da Segunda Avaliação:");
   scanf("%f", &n2);
   while((10<n2) || (n2<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10:");
         scanf("%f", &n2);
         getchar();
      }

   //N3
   printf("Insira a nota da Terceira Avaliação:");
   scanf("%f", &n3);
   while((10<n3) || (n3<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10:");
         scanf("%f", &n3);
         getchar();
      }

   //ME
   printf("Insira a media dos exercicios:");
   scanf("%f", &me);
   while((10<me) || (me<0))
      {
         printf("Valor Invalido! \n");
         printf("Insira um valor de 0 a 10:");
         scanf("%f", &me);
         getchar();
      }

   //Processamento
   ma = ((n1+n2*2+n3*3+me)/7);
   if(ma >= 9.0)
    {
      x = 'A';
    }
   if(( 9 > ma ) && ( ma >= 7.5 ))
    {
      x = 'B';
    }
   if((7.5>ma) && (ma>=6))
    {
      x = 'C';
    }
   if((6>ma) && (ma>=4))
    {
      x = 'D';
    }
   if(4.0>ma)
    {
      x = 'E';
    }
   //Saida de dados
   printf("Aluno   : %d\n",id);
   printf("Nota   1: %.2f \n",n1);
   printf("Nota   2: %.2f \n",n2);
   printf("Nota   3: %.2f \n",n3);
   printf("Nota  EX: %.2f \n",me);
   printf("Media de Aproveitamento: %.2f  ===> ", ma);
   printf("%c \n", x);
   if(ma>=6)
   {
      printf("Aprovado! \n");
   }
   else
   {
      printf("Reprovado! \n");
   }
   printf("Deseja continuar? (S/N)\n");

   scanf("%c", &zn);
   zn = getchar();


   }while ((zn != 'n') && (zn != 'N'));
   printf("\nfim \n");
   p=0;
}

File: 1º Semestre/Algoritmo/Lista1Full.kt
import java.util.*

fun main() {
    val teclado = Scanner(System.`in`)
    var continuar = true
    do{
        println("""
            Este programa contem todos os exercicios da lista 1 de Algoritmos.
            Ex 1 - Antecessor e Sucessor      |      Ex 7 - Calculo da media escolar
            Ex 2 - Reajuste +10%              |      Ex 8 - N > 50.000?
            Ex 3 - Desconto de 9%             |      Ex 9 - São multiplos?
            Ex 4 - Salario Professor          |      Ex 10 - inflação de Produtos
            Ex 5 - Carro Novo                 |      Ex 11 - Maior Numero
            Ex 6 - Reajuste Salarial          |      Ex 12 - Media + Conceito Escolar
            Digite o numero do exercicio que deseja executar:
        """.trimIndent())

        when (teclado.nextInt()){
            1 -> continuar = ex1(teclado)
            2 -> continuar = ex2(teclado)
            3 -> continuar = ex3(teclado)
            4 -> continuar = ex4(teclado)
            5 -> continuar = ex5(teclado)
            6 -> continuar = ex6(teclado)
            7 -> continuar = ex7(teclado)
            8 -> continuar = ex8(teclado)
            9 -> continuar = ex9(teclado)
            10 -> continuar = ex10(teclado)
            11 -> continuar = ex11(teclado)
            12 -> continuar = ex12(teclado)
            else -> println("Exercicio não encontrado")
        }
    }while (continuar)
    teclado.close()
}

fun ex1(teclado: Scanner): Boolean {
    println("Digite um numero inteiro: ")
    if (teclado.hasNextInt()){
        val numero = teclado.nextInt()
        println("O antecessor de $numero é ${numero-1} e o sucessor é ${numero+1}")
    } else {
        println("Valor invalido")
    }
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex2(teclado: Scanner): Boolean {
    println("Digite um valor para ser reajustado: ")
    if (teclado.hasNextDouble()){
        val valor = teclado.nextDouble()
        println("O valor reajustado é ${valor*1.1}")
    } else {
        println("Valor invalido")
    }
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun  ex3(teclado: Scanner): Boolean {
    println("Digite um valor para ser descontado: ")
    if (teclado.hasNextDouble()){
        val valor = teclado.nextDouble()
        println("O valor com desconto é ${valor*0.91}")
    } else {
        println("Valor invalido")
    }
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex4(teclado: Scanner): Boolean {
    println("Digite o valor da hora/aula: ")
    if (teclado.hasNextDouble()){
        val valor = teclado.nextDouble()
        println("Digite a quantidade de aulas dadas: ")
        if (teclado.hasNextInt()){
            val aulas = teclado.nextInt()
            println("Digite o percentual de desconto do INSS: ")
            if (teclado.hasNextDouble()){
                val desconto = teclado.nextDouble()
                println("O salario liquido do professor é ${valor*aulas*(1-desconto/100)}")
            } else {
                println("Valor invalido")
            }
        }
    }
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex5(teclado: Scanner): Boolean{
    var preco: Double
    var lucro: Double
    var imposto: Double
    println("Digite o preço de fabrica do carro: ")
    do {
        preco = teclado.nextDouble()
        if (preco <= 0){
            println("Valor invalido")
        }
    }while (preco <= 0)
    println("Digite o percentual de lucro do distribuidor: ")
    do {
        lucro = teclado.nextDouble()
        if (lucro <= 0){
            println("Valor invalido")
        }
    }while (lucro <= 0)
    println("Digite o percentual de impostos: ")
    do {
        imposto = teclado.nextDouble()
        if (imposto <= 0){
            println("Valor invalido")
        }
    }while (imposto <= 0)
    println("O preço final do carro é ${preco*(1+lucro/100)*(1+imposto/100)}")
    println("O valor do imposto é ${preco*(1+lucro/100)*imposto/100}")
    println("O valor do lucro do distribuidor é ${preco*lucro/100}")
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex6(teclado: Scanner): Boolean {
    var valor: Double
    println("Digite o valor do salario: ")
    do {
        valor = teclado.nextDouble()
        if (valor <= 0){
            println("Valor invalido")
        }
    }while (valor <= 0)
    if (valor < 800){
        println("O salario de R$${valor} foi reajustado para R$${valor*1.15}")
    } else if (valor > 1000){
        println("O salario de R$${valor} foi reajustado para R$${valor*1.05}")
    } else {
        println("O salario de R$${valor} foi reajustado para R$${valor*1.1}")
    }
    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex7(teclado: Scanner): Boolean {
    var n1: Double
    var n2: Double
    var n3: Double
    var n4: Double

    println("Digite a nota da primeira avaliação: ")
    do {
        n1 = teclado.nextDouble()
        if (n1 < 0 || n1 > 10){
            println("Valor invalido")
        }
    }while (n1 < 0 || n1 > 10)

    println("Digite a nota da segunda avaliação: ")
    do {
        n2 = teclado.nextDouble()
        if (n2 < 0 || n2 > 10){
            println("Valor invalido")
        }
    }while (n2 < 0 || n2 > 10)

    println("Digite a nota da terceira avaliação: ")
    do {
        n3 = teclado.nextDouble()
        if (n3 < 0 || n3 > 10) {
            println("Valor invalido")
        }
    }while (n3 < 0 || n3 > 10)

    println("Digite a nota da quarta avaliação: ")
    do {
        n4 = teclado.nextDouble()
        if (n4 < 0 || n4 > 10) {
            println("Valor invalido")
        }
    }while (n4 < 0 || n4 > 10)

    val media: Double = (n1+n2+n3+n4)/4
    println("A media do aluno é $media")
    if (media >= 6){
        println("O aluno foi aprovado")
    } else {
        println("O aluno foi reprovado")
    }

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex8(teclado: Scanner): Boolean {
    var n1: Double
    var n2: Double
    var n3: Double
    var n4: Double
    var n5: Double

    do{
        println("Digite um valor entre 10 e 25: ")
        do {
            n1 = teclado.nextDouble()
            if (n1 <= 10 || n1 >= 25) {
                println("Valor invalido")
            }
        } while (n1 <= 10 || n1 >= 25)

        println("Digite um valor positivo: ")
        do {
            n2 = teclado.nextDouble()
            if (n2 < 0) {
                println("Valor invalido")
            }
        } while (n2 < 0)

        n3 = n1 * n2
        n4 = n1 * n2 * n3
        n5 = (n1 * n1) + (n2 * n2) + (n3 * n3) + (n4 * n4)

        if (n5 < 50000) {
            println("O valor de $n5 é menor que 50.000")
            println("Por favor, digite novos valores")
            println("Pressione enter para continuar")
            teclado.nextLine()
        } else {
            println("O valor de $n5 é maior que 50.000")
            println("Pressione enter para continuar")
            teclado.nextLine()
        }
    }while (n5 < 50000)

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex9(teclado: Scanner): Boolean {
    var n1: Int
    var n2: Int

    do {
        println("Digite o primeiro numero: ")
        n1 = teclado.nextInt()
        if (n1 <= 0) {
            println("Valor invalido")
        }
    } while (n1 <= 0)

    do {
        println("Digite o segundo numero: ")
        n2 = teclado.nextInt()
        if (n2 <= 0) {
            println("Valor invalido")
        }
    } while (n2 <= 0)

    if (n1 % n2 == 0 || n2 % n1 == 0){
        println("Os numeros são multiplos")
    } else {
        println("Os numeros não são multiplos")
    }

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex10(teclado: Scanner): Boolean {
    var valor: Double

    do {
        println("Digite o valor do produto: ")
        valor = teclado.nextDouble()
        if (valor <= 0) {
            println("Valor invalido")
        }
    } while (valor <= 0)

    if (valor < 100){
        println("Preço reajustado: R$${valor*1.1}")
    }else{
        println("Preço reajustado: R$${valor*1.2}")
    }

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex11(teclado: Scanner): Boolean {

    println("Digite o primeiro numero: ")
    val n1: Int = teclado.nextInt()
    println("Digite o segundo numero: ")
    val n2: Int = teclado.nextInt()

    if (n1 > n2){
        println("$n1 > $n2")
    } else if (n1 < n2){
        println("$n2 > $n1")
    } else {
        println("$n1 = $n2")
    }

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}

fun ex12(teclado: Scanner): Boolean {
    var n1: Double
    var n2: Double
    var n3: Double
    var mediaExercicios: Double
    val conceito: String

    println("Digite sua ID: ")
    val id: Int = teclado.nextInt()

    do {
        println("Digite a nota da primeira avaliação: ")
        n1 = teclado.nextDouble()
        if (n1 < 0 || n1 > 10) {
            println("Valor invalido")
        }
    } while (n1 < 0 || n1 > 10)

    do {
        println("Digite a nota da segunda avaliação: ")
        n2 = teclado.nextDouble()
        if (n2 < 0 || n2 > 10) {
            println("Valor invalido")
        }
    } while (n2 < 0 || n2 > 10)

    do {
        println("Digite a nota da terceira avaliação: ")
        n3 = teclado.nextDouble()
        if (n3 < 0 || n3 > 10) {
            println("Valor invalido")
        }
    } while (n3 < 0 || n3 > 10)

    do {
        println("Digite a media dos exercicios: ")
        mediaExercicios = teclado.nextDouble()
        if (mediaExercicios < 0 || mediaExercicios > 10) {
            println("Valor invalido")
        }
    } while (mediaExercicios < 0 || mediaExercicios > 10)

    val mediaAproveitamento: Double = (n1 + n2 * 2 + n3 * 3 + mediaExercicios) / 7

    conceito = if (mediaAproveitamento >= 9){
        "A"
    } else if (mediaAproveitamento >= 7.5){
        "B"
    } else if (mediaAproveitamento >= 6){
        "C"
    } else if (mediaAproveitamento >= 4){
        "D"
    } else {
        "E"
    }

    println("ID: $id")
    println("Nota da primeira avaliação: $n1")
    println("Nota da segunda avaliação: $n2")
    println("Nota da terceira avaliação: $n3")
    println("Media dos exercicios: $mediaExercicios")
    println("Media de aproveitamento: $mediaAproveitamento")
    println("Conceito: $conceito")

    when (conceito){
        "A", "B", "C" -> println("APROVADO")
        else -> println("REPROVADO")
    }

    println("Deseja voltar ao menu? (S/N)")
    return teclado.next().equals("s", true)
}
